testsuite/tests/singlecluster/extensions/plan_policy/test_plan_policy.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for PlanPolicy functionality including authentication and rate limiting."""
+
+import pytest
+
+from testsuite.httpx.auth import HttpxOidcClientAuth
+from testsuite.kuadrant.extensions.plan_policy import PlanPolicy, Plan
+
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.authorino, pytest.mark.extensions]
+
+PLANS = {
+    "gold": Plan(
+        tier="gold",
+        predicate='has(auth.identity) && auth.identity.planID == "gold"',
+        limits={"custom": [{"limit": 3, "window": "20s"}]},
+    ),
+    "silver": Plan(
+        tier="silver",
+        predicate='has(auth.identity) && auth.identity.planID == "silver"',
+        limits={"custom": [{"limit": 2, "window": "20s"}]},
+    ),
+    "bronze": Plan(
+        tier="bronze",
+        predicate='has(auth.identity) && auth.identity.planID == "bronze"',
+        limits={"custom": [{"limit": 1, "window": "20s"}]},
+    ),
+}
+
+
+@pytest.fixture(scope="module")
+def target(request):
+    """Returns the test target(gateway or route)"""
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture(scope="module")
+def realm_with_attributes(keycloak):
+    """
+    Ensure the realm allows a custom user attribute and configure client mapper.
+    """
+    keycloak.realm.add_user_attributes("planID", "Plan ID")
+    keycloak.client.add_user_attribute_mapper("planID", "planID")
+
+
+@pytest.fixture(scope="module")
+def user_with_gold_plan(keycloak, realm_with_attributes, blame):  # pylint: disable=unused-argument
+    """
+    Creates new user with "gold" plan.
+    """
+    user = keycloak.realm.create_user(blame("someuser"), blame("password"), attributes={"planID": ["gold"]})
+    return HttpxOidcClientAuth.from_user(keycloak.get_token, user=user)
+
+
+@pytest.fixture(scope="module")
+def user_with_silver_plan(keycloak, realm_with_attributes, blame):  # pylint: disable=unused-argument
+    """
+    Creates new user with "silver" plan.
+    """
+    user = keycloak.realm.create_user(blame("someuser"), blame("password"), attributes={"planID": ["silver"]})
+    return HttpxOidcClientAuth.from_user(keycloak.get_token, user=user)
+
+
+@pytest.fixture(scope="module")
+def user_with_bronze_plan(keycloak, realm_with_attributes, blame):  # pylint: disable=unused-argument
+    """
+    Creates new user with "bronze" plan.
+    """
+    user = keycloak.realm.create_user(blame("someuser"), blame("password"), attributes={"planID": ["bronze"]})
+    return HttpxOidcClientAuth.from_user(keycloak.get_token, user=user)
```

Comment:
all these fixtures are only different in their planID name. You think you can parameterize this fixture directly, instead of parametrizing the getfixture parameter inside the test?